### PR TITLE
Bgd 4180 document memory metrics

### PR DESCRIPTION
**Pull request by crezvoy**
# Jira Ticket

[BGD-4180](https://spotinst.atlassian.net/browse/BGD-4180)
[DOC-1555](https://spotinst.atlassian.net/browse/DOC-1555)

# Demo

Screenshot 2023-11-08 at 18 06 03

Screenshot 2023-11-08 at 18 07 21

[BGD-4180]: https://spotinst.atlassian.net/browse/BGD-4180?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DOC-1555]: https://spotinst.atlassian.net/browse/DOC-1555?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by originou — COMMENTED** [review]
lgfm but better to have another eyes

File: src/docs/ocean-spark/configure-spark-apps/memory-&-cores.md [hunk 1]
```diff
@@ -104,6 +136,31 @@ Be careful when entering memory settings manually, as it is easy to make mistake
 
 If you would like to investigate some of these configurations further, the official Apache Spark documentation page on [Running Spark on Kubernetes](https://spark.apache.org/docs/latest/running-on-kubernetes.html) contains useful information.
 
+### Memory autotuning strategies
+
+In addition to hard-coded memory string, two strategies are available to
+dynamically adjust the executor memory . These strategies analyze previous
```
Comment (thorsteinnth) [inline src/docs/ocean-spark/configure-spark-apps/memory-&-cores.md:142]: extra space before the "." here

File: src/docs/ocean-spark/product-tour/view-application-details.md [hunk 2]
```diff
@@ -36,9 +36,28 @@ The App Info area gives you a quick point of reference for vital information abo
...
+The Insights area gives information about the resource usage of the application
+over time. The first tab shows executor CPU usage, broken down by categories
+(CPU, I/O, shuffle, GC, Spark internals). This graph aligns with a timeline of
+your Spark jobs and stages, so that it's easy to correlate CPU metrics with the
+code of your Spark application.
+
+<img src="/ocean-spark/_media/view-app-details-04.png" />
+
+On the second tab, you get a report of the memory usage of your spark executors
+over the application's job and stages timeline. on the left hand side, you can
```
Comment (thorsteinnth) [inline src/docs/ocean-spark/product-tour/view-application-details.md:50]: on -> On

File: src/docs/ocean-spark/configure-spark-apps/memory-&-cores.md [hunk 3]
```diff
@@ -104,6 +136,31 @@ Be careful when entering memory settings manually, as it is easy to make mistake
...
+automatically bump the memory request up (doubling it by default) if the
+previous app with the same job name failed with OOM (Out Of Memory) errors.
+
+```json
+{ "executor": { "strategy": "oomRecovery" } }
+```
+
+The autotuning strategy expands on the oomRecovery strategy by additionally
+adjusting memory request down to match the observed memory usage of previous
+apps. The autotuning strategy is only available for spark 3 apps as it relies on
```
Comment (thorsteinnth) [inline src/docs/ocean-spark/configure-spark-apps/memory-&-cores.md:155]: spark -> Spark

File: src/docs/ocean-spark/configure-spark-apps/memory-&-cores.md (hunk 1 above)
Comment (crezvoy) [reply src/docs/ocean-spark/configure-spark-apps/memory-&-cores.md]: 👍

File: src/docs/ocean-spark/product-tour/view-application-details.md (hunk 2 above)
Comment (crezvoy) [reply src/docs/ocean-spark/product-tour/view-application-details.md]: 👍

File: src/docs/ocean-spark/configure-spark-apps/memory-&-cores.md (hunk 3 above)
Comment (crezvoy) [reply src/docs/ocean-spark/configure-spark-apps/memory-&-cores.md]: 👍

**Pull request merged**